diff --git a/Cargo.toml b/Cargo.toml
index b8b825f..952e25c 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -9,7 +9,7 @@ name = "wassima"
 crate-type = ["cdylib"]
 
 [dependencies]
-pyo3 = { version = "0.23.1", features = ["abi3-py37", "extension-module"] }
+pyo3 = { version = "0.23.1", features = ["extension-module"] }
 rustls-native-certs = "0.7.3"
 
 [package.metadata.maturin]
diff --git a/pyproject.toml b/pyproject.toml
index 60bb86b..524b1b6 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -46,7 +46,7 @@ dependencies = [
 "Issue tracker" = "https://github.com/jawah/wassima/issues"
 
 [tool.maturin]
-features = ["pyo3/extension-module", "pyo3/abi3-py37"]
+features = ["pyo3/extension-module"]
 module-name = "wassima._rustls"
 
 [tool.pytest.ini_options]
diff --git a/src/lib.rs b/src/lib.rs
index fe02367..75cdafa 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -20,7 +20,7 @@ fn root_der_certificates(py: Python) -> PyResult<Vec<Bound<'_, PyBytes>>> {
     Ok(roots)
 }
 
-#[pymodule]
+#[pymodule(gil_used = false)]
 fn _rustls(_py: Python, m: &Bound<'_, PyModule>) -> PyResult<()> {
     m.add_function(wrap_pyfunction!(root_der_certificates, m)?)?;
     Ok(())
